packages/nextjs-mf/src/plugins/ChildFederationPlugin.ts

```diff
@@ -0,0 +1,362 @@
+    extraOptions: NextFederationPluginExtraOptions
+  ) {
+    this._options = options;
+    this._extraOptions = extraOptions;
+  }
+
+  apply(compiler: Compiler) {
+    const webpack = compiler.webpack;
+    const LibraryPlugin = webpack.library.EnableLibraryPlugin;
+    const LoaderTargetPlugin = webpack.LoaderTargetPlugin;
+    const library = compiler.options.output.library;
+    const isServer = compiler.options.name === 'server';
+    const isDev = compiler.options.mode === 'development';
+    let outputPath: string;
+
+    if (isDev && isServer) {
+      outputPath = path.join(getOutputPath(compiler), 'static/ssr');
+    } else {
+      if (isServer) {
+        outputPath = path.join(getOutputPath(compiler), 'static/ssr');
+      } else {
+        outputPath = compiler.options.output.path as string;
+      }
+    }
+
+    compiler.hooks.thisCompilation.tap(CHILD_PLUGIN_NAME, (compilation) => {
+      let plugins;
+      const buildName = this._options.name;
+      // using ModuleFederationPlugin does not work, i had to fork because of afterPlugins hook on containerPlugin.
+      const FederationPlugin = ChildFriendlyModuleFederationPlugin;
+
+      const childOutput = {
+        ...compiler.options.output,
+        path: outputPath,
+        // path: deriveOutputPath(isServer, compiler.options.output.path),
+        publicPath: 'auto',
+        chunkLoadingGlobal: buildName + 'chunkLoader',
+        uniqueName: buildName,
+        library: {
+          name: buildName,
+          type: library?.type as string,
+        },
+        chunkFilename: (
+          compiler.options.output.chunkFilename as string
+        )?.replace('.js', '-fed.js'),
+        filename: (compiler.options.output.filename as string)?.replace(
+          '.js',
+          '-fed.js'
+        ),
+      };
+
+      const federationPluginOptions: ModuleFederationPluginOptions = {
+        // library: {type: 'var', name: buildName},
+        ...this._options,
+        filename: computeRemoteFilename(
+          isServer,
+          this._options.filename as string
+        ),
+        exposes: {
+          ...this._options.exposes,
+          ...(this._extraOptions.exposePages
+            ? exposeNextjsPages(compiler.options.context as string)
+            : {}),
+        },
+        runtime: false,
+        shared: {
+          ...(this._extraOptions.skipSharingNextInternals
+            ? {}
+            : externalizedShares),
+          ...this._options.shared,
+        },
+      };
+
+      if (compiler.options.name === 'client') {
+        plugins = [
+          new FederationPlugin(federationPluginOptions),
+          // FIXME: JsonpTemplate no more expects constructor arguments
+          // https://github.com/webpack/webpack/blob/main/lib/web/JsonpTemplatePlugin.js
+          // Removing 'childOutput' as TS is throwing an error
+          new webpack.web.JsonpTemplatePlugin(), // new webpack.web.JsonpTemplatePlugin(childOutput),
```

Comment:
@ScriptedAlchemy `JsonpTemplatePlugin` isn't accepting any constructor arguments.

I'm not sure how we can `childOutput` to TemplatePlugin.

**REF:** https://github.com/webpack/webpack/blob/main/lib/web/JsonpTemplatePlugin.js

Reply:
I have checked this Plugin on both `webPack 5.64.4 & latest`, they both seems to not have any constructor arguments.